rootfs/etc/nginx/lua/util/defer.lua

```diff
@@ -0,0 +1,57 @@
+local util = require("util")
+
+local _M = {
+  timer_started = false,
+  queue = {},
+  MAX_QUEUE_SIZE = 10000
```

Comment:
none of these three variables need to be exported, you can define them as local

Reply:
But don't we want `timer_started` and `queue` to be accessible by multiple requests?

E.g. if `timer_started` is set to true in the global scope by a request, and it hasn't hit the timer phase yet, during which time another request invokes `logging.call()`, the expectation is that the second request simply adds its work to the queue instead of creating another timer no?

Reply:
I agree with you for MAX_QUEUE_SIZE though - stays constant 👍

Reply:
> But don't we want timer_started and queue to be accessible by multiple requests?

`local` variables defined at module scope **are** accessible by multiple requests. They're effectively global variables to the NGINX worker. NGINX is more or less single threaded, and the timer phase executes outside the context of any request. So in the example you have above, it is possible for separate requests to enqueue work to the same queue.

Reply:
Thank you @csfrancis! Cleared this up with Elvin over Slack but thank you for the context nonetheless :)